### DISCUSSION TITLE
Can't connect to ADB

### DISCUSSION THREAD

**Discussion by loudonadam** [AdriaGual/marvel-snap-bot#36] [Q&A] [answered]
I'm trying to set up the ADB bit of the process, but am having difficulty. I couldn't find any sort of platform-tools folder when I was using Bluestacks, so I installed Nox to try and get it working that way. Unforunately, I still can't find it here. I found a folder here (C:\Program Files (x86)\Nox\bin) that has an adb.exe file in it. If I right click in that folder, launched terminal and then do "adb devices" I get an error. That tells me to do ".\adb", so if I do ".\adb devices" I get the response 
`List of devices attached
127.0.0.1:62001 device`

But no emulator (note, Marvel Snap is running on Nox at the time I executed this). What am I doing wrong here. And do I need to change the 127 number to this number somewhere in the config?

Any help would be greatly appreciated.

**Comment by AdriaGual** [comment, marked as answer]
Ok, so instead of running Marvel Snap on Nox, you have to run it on Bluestacks.

And put your path to the adb on the `config.py` file with double slashes `\\`

**Reply by loudonadam** [reply]
I can't find the adb filepath in Bluestacks. I enabled the feature and rebooted, but can't find any platform-tools folder.

**Reply by AdriaGual** [reply]
The platform-tools folder isn't from BS, it's from the proper android sdk, you have to install it and it will be in a path similar to:
`%USERPROFILE%\AppData\Local\Android\sdk\platform-tools`

**Reply by loudonadam** [reply]
Oh, ok. So do I need to install Android Studio in order to install the Android SDK?

**Reply by AdriaGual** [reply]
You can install only the command line tools if you prefer, just make sure that in the installed folder you can execute the command `adb devices`.

https://developer.android.com/studio#command-tools
